### PR TITLE
Adds Support for SQL Server Dialect

### PR DESCRIPTION
**Pull request by dylel**
Not entirely sure how the type mapping was supposed to work but i've added a mapping that seems to work correctly. Can't add tests for kysely yet as the sql server dialect hasn't been merged.

### CODE REVIEW

**Comment by valtyr** [comment]
Hey @dylel, thank you so much for the pull request! I'll keep my eyes peeled for when the PR gets merged on Kysely's side. Once that's done we can add a test script to validate that the types are correct.

I'm assuming this is the PR you're talking about: https://github.com/kysely-org/kysely/pull/595

**Comment by dylel** [comment]
Yeah that's the one I was referring to, anything else you would like me to do so this can be merged?

**Comment by valtyr** [comment]
As I say I'd prefer not merging this before I can validate that the type map is totally correct. If you add a changeset using `yarn changeset` I can generate a snapshot release that you and others can use in the meantime.

**Comment by dylel** [comment]
No worries, i've just installed it straight from the pr for the time being

``` pnpm install valtyr/prisma-kysely#refs/pull/68/head```

**Comment by fehnomenal** [comment]
Thank you for this PR, @dylel.

The kysely PR is merged but not yet released @valtyr. Can we still progress on this PR?

**Comment by thedrewguy** [comment]
Kysely has released sql server support. Would be great to see this PR merged and released!

**Comment by fehnomenal** [comment]
Yes please. For the record: I already use this PR as a patch in production.

**Comment by valtyr** [comment]
Hey there, sorry for the delay. I'll merge this now.

**Comment by jrvgr** [comment]
Any updates?

File: src/helpers/generateFieldType.ts [hunk 1]
```diff
@@ -52,6 +52,19 @@ export const postgresqlTypeMap: TypeMap = {
   Unsupported: "unknown",
 };
 
+export const sqlServerTypeMap: TypeMap = {
+  BigInt: "number",
+  Boolean: "boolean",
+  Bytes: "Buffer",
+  DateTime: "Timestamp",
+  Decimal: "string",
```
Comment (fehnomenal) [inline src/helpers/generateFieldType.ts:60]: The same as in #80 

```suggestion
  Decimal: "number",
```

Comment (valtyr) [reply src/helpers/generateFieldType.ts:60]: @fehnomenal I've closed #80 due to default driver behavior. I'm going to say that this change is not needed for now but I'll add a dialect test later to be sure.

Comment (fehnomenal) [reply src/helpers/generateFieldType.ts:60]: Well, alright I guess you're right. In the meantime I noticed that I can override the generated types via the config.

**Pull request merged**